clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -487,7 +487,12 @@ else if (strategy == OffsetResetStrategy.LATEST)
                 } else {
                     TopicPartition partition = nextInLineRecords.partition;
                     fetchedPartition = partition;
-                    fetchedOffsets = subscriptions.position(partition);
+                    if (subscriptions.isAssigned(partition)) {
```

Comment:
Mentioned offline, but a better solution (if Becket is right that the try/catch needs to cover this branch) would be the following:
```java
fetchedOffsets = nextInLineRecords.fetchOffset;
```
Also, seems this variable shouldn't be plural.